praw/models/reddit/modmail.py

```diff
@@ -70,7 +70,8 @@ def __init__(self, reddit, id=None,  # pylint: disable=redefined-builtin
         """Construct an instance of the ModmailConversation object."""
         super(ModmailConversation, self).__init__(reddit, _data)
 
-        self.id = id  # pylint: disable=invalid-name
+        if id is not None:
```

Comment:
If it actually requires fetching, then yes an integration test makes sense. I might have misinterpreted what was happening.